### PR TITLE
saving new translation

### PR DESCRIPTION
**Pull request by pavel-kru**
1. input was changed with contentEditable div
2. data updating logic added
3. some refactoring was made due to updates
4. some changes in css

### CODE REVIEW

File: src/components/csvLocalizer/jsonLocalizer.js [hunk 1]
```diff
@@ -23,12 +23,12 @@ const JsonLocalizer = () => {
...
   const [activeCellId, setActiveCellId] = React.useState('');
-  const [newTranslation, setNewtranslation] = React.useState('');
+  const [alert, setAlert] = React.useState('');
 
-  const inputRef = React.useRef();
+  const textareaRef = React.useRef();
 
   React.useEffect(() => {
-    inputRef.current?.focus();
+    textareaRef.current?.focus();
```
Comment (Pavel-Yarakhovich) [inline src/components/csvLocalizer/jsonLocalizer.js:31]: Will it try to focus the textarea when the `activeCellId` is set to empty string?

File: src/components/csvLocalizer/jsonLocalizer.js [hunk 2]
```diff
@@ -48,14 +48,32 @@ const JsonLocalizer = () => {
 
   const onCancel = React.useCallback(() => {
     setActiveCellId('');
-    setNewtranslation('');
+    setAlert('');
   }, []);
 
+  const activeCellContent =
+    activeCellId &&
+    data[activeCellId.split('-')[0]][+activeCellId.split('-')[1]];
```
Comment (Pavel-Yarakhovich) [inline src/components/csvLocalizer/jsonLocalizer.js]: This line is hardly readable.
I'm sure you know what to expect from the activeCellId, but first what I thought of was 'what if this expression will return undefined? we'll get the activeCellContent equal to the activeCellId'.
let activeCellContent;
const [x, y] = activeCellId?.split('-');
if (x && y) activeCellContent = data[x][y];

File: src/components/csvLocalizer/jsonLocalizer.js [hunk 3]
```diff
@@ -48,14 +48,32 @@ const JsonLocalizer = () => {
...
+
   const saveTranslation = React.useCallback(() => {
-    console.log(`id: ${activeCellId}`, `translation: ${newTranslation}`);
+    if (activeCellContent === textareaRef.current.innerText) {
+      setAlert('Please, edit a translation or cancel');
+      return;
+    }
+    const [key, index] = activeCellId.split('-');
```
Comment (Pavel-Yarakhovich) [inline src/components/csvLocalizer/jsonLocalizer.js]: Good!

File: src/components/csvLocalizer/jsonLocalizer.js [hunk 4]
```diff
@@ -48,14 +48,32 @@ const JsonLocalizer = () => {
...
+    setAlert('');
   }, []);
 
+  const activeCellContent =
+    activeCellId &&
+    data[activeCellId.split('-')[0]][+activeCellId.split('-')[1]];
+
   const saveTranslation = React.useCallback(() => {
-    console.log(`id: ${activeCellId}`, `translation: ${newTranslation}`);
+    if (activeCellContent === textareaRef.current.innerText) {
```
Comment (Pavel-Yarakhovich) [inline src/components/csvLocalizer/jsonLocalizer.js]: Does the `textareaRef.current` always exist at this moment?

File: src/components/csvLocalizer/jsonLocalizer.js [hunk 5]
```diff
@@ -74,6 +92,7 @@ const JsonLocalizer = () => {
             .filter(Boolean)
             .map(([key, values]) => {
               const cellWidth = 100 / values.length;
+              const showTextArea = activeCellId?.split('-')[0] === key;
```
Comment (Pavel-Yarakhovich) [inline src/components/csvLocalizer/jsonLocalizer.js]: We use the same value in at least three places `activeCellId?.split('-')[0]`. Can we keep it in state?

**Review by Pavel-Yarakhovich — CHANGES_REQUESTED** [review]

File: src/components/csvLocalizer/jsonLocalizer.js (hunk 1 above)
Comment (pavel-kru) [reply src/components/csvLocalizer/jsonLocalizer.js:31]: Yes, it works fine with empty string.

File: src/components/csvLocalizer/jsonLocalizer.js (hunk 4 above)
Comment (pavel-kru) [reply src/components/csvLocalizer/jsonLocalizer.js]: Yes, at this moment it exists and has innerText equal to content of active cell or it's own.

File: src/components/csvLocalizer/jsonLocalizer.js (hunk 5 above)
Comment (pavel-kru) [reply src/components/csvLocalizer/jsonLocalizer.js]: edited

File: src/components/csvLocalizer/jsonLocalizer.js (hunk 2 above)
Comment (pavel-kru) [reply src/components/csvLocalizer/jsonLocalizer.js]: edited

File: src/components/csvLocalizer/jsonLocalizer.js (hunk 3 above)
Comment (pavel-kru) [reply src/components/csvLocalizer/jsonLocalizer.js]: Thanks!

**Pull request merged**